sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadSupport.scala

```diff
@@ -88,7 +88,7 @@ class ParquetReadSupport(
       SQLConf.NESTED_SCHEMA_PRUNING_ENABLED.defaultValue.get)
     val parquetFileSchema = context.getFileSchema
     val parquetClippedSchema = ParquetReadSupport.clipParquetSchema(parquetFileSchema,
```

Comment:
IIUC `clipParquetSchema` is not designed to check schema mismatch. We should put the check in the readers. It's also better to do so, as reader implementation decides how the check should be done.

I opened a PR to do it: https://github.com/apache/spark/pull/31347 . Feel free to incorporate it in your PR.

Reply:
It's great. Sure. I'll incorporate it here.